From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: BuildTools <46540330+willkroboth@users.noreply.github.com>
Date: Fri, 19 Aug 2022 16:11:51 -0400
Subject: [PATCH] Add method isTickingWorlds() to Bukkit to allow plugins to
 know when it is safe to attempt loading or unloading worlds.


diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index 840aaf9e8fc828b5a7ea02252038c6524680f2e0..b2d75b65a5067f0dfe5d447fd9f28aa7fbce9077 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -753,6 +753,17 @@ public final class Bukkit {
         return server.getWorlds();
     }
 
+    /**
+     * Returns true if the worlds are being ticked, and false otherwise.
+     * If the worlds are being ticked, it is not currently safe to create or load a world,
+     * so it is recommended that plugins check this value before attempting to do so.
+     *
+     * @return if the worlds are being ticked.
+     */
+    public static boolean isTickingWorlds(){
+        return server.isTickingWorlds();
+    }
+
     /**
      * Creates or loads a world with the given name using the specified
      * options.
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index da5cab4246bd253fcc4e4d9574bdae1867ebb5ab..dd37fe7ed8a016bdfbd165c913d53169f5ed3a2d 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -622,6 +622,15 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
     @NotNull
     public List<World> getWorlds();
 
+    /**
+     * Returns true if the worlds are being ticked, and false otherwise.
+     * If the worlds are being ticked, it is not currently safe to create or load a world,
+     * so it is recommended that plugins check this value before attempting to do so.
+     *
+     * @return if the worlds are being ticked.
+     */
+    public boolean isTickingWorlds();
+
     /**
      * Creates or loads a world with the given name using the specified
      * options.
